services/core/shared/core.js

```diff
@@ -326,27 +335,123 @@ const getBlocks = async params => {
+	return tx;
+};
+
+const calculateBlockSize = block => {
+	let blockSize = 0;
+	if (block.numberOfTransactions === 0) return blockSize;
+
+	const payload = block.transactions.data;
+	payload.forEach(transaction => {
+		const tx = getTransactionInstanceByType(transaction);
+		const transactionSize = tx.getBytes().length;
+		blockSize += transactionSize;
+	});
+
+	return blockSize;
+};
+
+const calculateWeightedAvg = blocks => {
+	const blockSizes = [];
+
+	blocks.forEach(block => blockSizes.push(calculateBlockSize(block)));
+
+	const decayFactor = config.feeEstimates.wavgDecayPercentage / 100;
+	let weight = 1;
+	const wavgLastBlocks = blockSizes.reduce((a, b) => {
+		weight *= 1 - decayFactor;
+		return a + (b * weight);
+	});
+
+	return wavgLastBlocks;
+};
+
+const calulateAvgFeePerByte = (mode, blockSize, transactionDetails) => {
+	if (blockSize === 0) return 0;
+
+	const allowedModes = ['med', 'high'];
+	const lowerPercentile = allowedModes.includes(mode) && mode === 'med'
+		? config.feeEstimates.medEstLowerPercentile : config.feeEstimates.highEstLowerPercentile;
+	const upperPercentile = allowedModes.includes(mode) && mode === 'med'
+		? config.feeEstimates.medEstUpperPercentile : config.feeEstimates.highEstUpperPercentile;
+	const lowerBytePos = Math.ceil((lowerPercentile / 100) * blockSize);
+	const upperBytePos = Math.floor((upperPercentile / 100) * blockSize);
+
+	let currentBytePos = 0;
+	let totalFeePriority = 0;
+	transactionDetails.forEach(transaction => {
+		if (currentBytePos <= lowerBytePos && lowerBytePos < currentBytePos + transaction.size
+			&& currentBytePos + transaction.size <= upperBytePos) {
+			totalFeePriority += transaction.feePriority
+				* (currentBytePos + transaction.size - lowerBytePos + 1);
+		}
+
+		if (lowerBytePos <= currentBytePos && currentBytePos + transaction.size <= upperBytePos) {
+			totalFeePriority += transaction.feePriority * transaction.size;
+		}
+
+		if (lowerBytePos <= currentBytePos && upperBytePos <= currentBytePos + transaction.size) {
+			totalFeePriority += transaction.feePriority * (upperBytePos - currentBytePos + 1);
+		}
+
+		currentBytePos += transaction.size;
+	});
+
+	const avgFeePriority = totalFeePriority / (upperBytePos - lowerBytePos + 1);
+	return avgFeePriority;
+};
+
+const calculateFeePerByte = block => {
+	const feePerByte = {};
+	const payload = block.transactions.data;
+	const transactionDetails = [];
+
+	payload.forEach(transaction => {
+		const tx = getTransactionInstanceByType(transaction);
+		const transactionSize = tx.getBytes().length;
+		const { minFee } = tx;
+		const feePriority = (transaction.fee - minFee) / transactionSize;
+		transactionDetails.push({
+			id: transaction.id,
+			size: transactionSize,
+			feePriority,
+		});
+	});
+	transactionDetails.sort((t1, t2) => t1.feePriority - t2.feePriority);
+
+	const blockSize = calculateBlockSize(block);
+
+	feePerByte.low = (blockSize < 12.5 * 2 ** 10) ? 0 : transactionDetails[0].feePriority;
```

Comment:
I see that the implementation assumes 1 KB as 1024 Bytes. I think rest of applications and protocol we have the same assumption but we should be sure of this for consistency.

Reply:
I confirmed it with @shuse2  to be 1024 KB.